llama-index-core/llama_index/core/callbacks/token_counting.py

```diff
@@ -53,11 +53,14 @@ def get_llm_token_counts(
             response_tokens = 0
 
             if response is not None and response.raw is not None:
-                usage = response.raw.get("usage", None)
+                if isinstance(response.raw, dict):
+                    raw_dict = response.raw
+                else:
+                    raw_dict = response.raw.model_dump()
```

Comment:
I think it might be safer to do `dict(response.raw)` ? or nah?

Reply:
I think it's less safe, as dict() does not respect custom serialization logic, nor does it convert the sub-models (usage). Do we expect anything else as input here besides dict or pydantic models?